kamon-akka-remote-2.4.x/src/main/scala/kamon/akka/instrumentation/RemotingInstrumentation.scala

```diff
@@ -92,13 +165,15 @@ class RemotingInstrumentation {
   @Around("decodeRemoteMessage(bs, provider, localAddress)")
   def aroundDecodeRemoteMessage(pjp: ProceedingJoinPoint, bs: ByteString, provider: RemoteActorRefProvider, localAddress: Address): AnyRef = {
     val ackAndEnvelope = AckAndTraceContextAwareEnvelopeContainer.parseFrom(bs.toArray)
-
     if (ackAndEnvelope.hasEnvelope && ackAndEnvelope.getEnvelope.hasTraceContext) {
       val remoteCtx = ackAndEnvelope.getEnvelope.getTraceContext
-      val ctx = Kamon.contextCodec().Binary.decode(
-        ByteBuffer.wrap(remoteCtx.getContext.toByteArray)
-      )
-      Kamon.storeContext(ctx)
+
+      if(remoteCtx.getContext.size() > 0) {
+        val ctx = Kamon.contextCodec().Binary.decode(
+          ByteBuffer.wrap(remoteCtx.getContext.toByteArray)
+        )
+        Kamon.storeContext(ctx)
```

Comment:
I just checked and I think even though this leaves the thread dirty, the actor instrumentation will clean it up afterwards but take a look, just in case to ensure there is no funny behavior around this.